### ISSUE TITLE
linter-python stopped returning errors after updating from 2.1.7 to 3.0.4

### ISSUE THREAD

**Issue by eguiraud** [pchomik/linter-python#27]
Uninstalling the latest version and re-installing 2.1.7 reverts the package to its correct behaviour.

When I say stopped returning errors I mean that, while working, it is as if everything I write is correct ("no issues is displayed..as if :P).

Enabling debug for this package and checking the console, this is the output (for a file with several trivial errors that can be found by pylama and pyflakes when run from commandline):

![bug](https://cloud.githubusercontent.com/assets/10999034/17135320/850caedc-532f-11e6-8450-685b8bcb6ede.png)

**Comment by pchomik** [comment]
"RAW OUTPUT" is an output from pylama application. Tool returns empty string - no errors.
Is it possible to see foo.py script? Could you change plugin setting "Lint Trigger" to "File saved or modified"?

**Comment by eguiraud** [comment]
Here are some more screenshots, taken right now, at seconds of distance, on a Dell XPS 13 laptop with an up-to-date installation of Ubuntu 14.04. I use i3 window manager.
`Atom --version` gives

```
Atom    : 1.8.0
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```

The output of `which pylama` corresponds to the value in the "Executable Path" setting of linter-python.

linter-python settings:
![bugoptions](https://cloud.githubusercontent.com/assets/10999034/17167768/22d2749e-53df-11e6-973c-698a30938adf.png)

foo.py and developer console showing no errors were found:
![bugoutput](https://cloud.githubusercontent.com/assets/10999034/17167778/2cb5f08a-53df-11e6-8ba5-fea84543d8a7.png)

manual calls to pylama and pyflakes from command line returning the correct list of errors:
![bugcorrectoutput](https://cloud.githubusercontent.com/assets/10999034/17167788/39046ca4-53df-11e6-88e3-d39f72b6a525.png)

**Comment by pchomik** [comment]
Thanks for all details. I will check what is going on.

**Comment by pchomik** [comment]
I'm not able to reproduce this error in my environment:
Test File:
![image](https://cloud.githubusercontent.com/assets/1779381/17204089/af94997e-54a3-11e6-8140-e35a2915a5c3.png)

Debug Log:
![image](https://cloud.githubusercontent.com/assets/1779381/17204077/9d10bcba-54a3-11e6-836d-014f8027bf6a.png)

In addition please execute following commands and provide results:

```
/home/blue/anaconda2/bin/pylama -l pyflakes --sort E,W,D -f pep8 /home/blue/foo.py
pylama --version
pyflakes --version
python --version
pip list
```

**Comment by eguiraud** [comment]
We might be onto something here:

```
bluehood:~$ /home/blue/anaconda2/bin/pylama -l pyflakes --sort E,W,D -f pep8 /home/blue/foo.py
usage: pylama [-h] [--verbose] [--version] [--format {pep8,pylint}]
              [--select SELECT] [--linters LINTERS] [--ignore IGNORE]
              [--skip SKIP] [--report REPORT] [--hook] [--async]
              [--options OPTIONS]
              [path]
pylama: error: unrecognized arguments: --sort /home/blue/foo.py
```

The other outputs:

```
bluehood:~$ pylama --version
pylama 3.0.2
bluehood:~$ pyflakes --version
1.0.0
bluehood:~$ python --version
Python 2.7.11 :: Anaconda 2.5.0 (64-bit)
bluehood:~$ pip list
abstract-rendering (0.5.1)
alabaster (0.7.7)
anaconda-client (1.2.2)
argcomplete (1.0.0)
astroid (1.4.5)
astropy (1.1.1)
Babel (2.2.0)
backports-abc (0.4)
backports.ssl-match-hostname (3.4.0.2)
beautifulsoup4 (4.4.1)
bitarray (0.8.1)
blaze (0.9.0)
bokeh (0.11.0)
boto (2.39.0)
Bottleneck (1.0.0)
cdecimal (2.3)
cffi (1.2.1)
clyent (1.2.0)
colorama (0.3.6)
conda (4.1.5)
conda-build (1.19.0)
conda-env (2.5.0a0)
configobj (5.0.6)
cryptography (0.9.1)
cycler (0.9.0)
Cython (0.23.4)
cytoolz (0.7.5)
datashape (0.5.0)
decorator (4.0.9)
docutils (0.12)
dynd (f641248)
enum34 (1.1.2)
et-xmlfile (1.0.1)
fastcache (1.0.2)
flake8 (2.5.4)
flaky (3.1.1)
Flask (0.10.1)
funcsigs (0.4)
future (0.15.2)
futures (3.0.3)
gevent (1.0.2)
gevent-websocket (0.9.5)
greenlet (0.4.9)
grin (1.2.1)
h5py (2.5.0)
idna (2.0)
ipaddress (1.0.14)
ipykernel (4.3.1)
ipython (4.1.2)
ipython-genutils (0.1.0)
ipywidgets (4.1.1)
itsdangerous (0.24)
jdcal (1.2)
jedi (0.9.0)
Jinja2 (2.8)
jsonschema (2.4.0)
jupyter (1.0.0)
jupyter-client (4.2.1)
jupyter-console (4.1.1)
jupyter-core (4.1.0)
lazy-object-proxy (1.2.1)
linecache2 (1.0.0)
llvmlite (0.8.0)
lxml (3.5.0)
MarkupSafe (0.23)
matplotlib (1.5.1)
mccabe (0.4.0)
mistune (0.7.1)
mpi4py (2.0.0)
multipledispatch (0.4.8)
nbconvert (4.1.0)
nbformat (4.0.1)
networkx (1.11)
nltk (3.1)
nose (1.3.7)
notebook (4.1.0)
numba (0.23.1)
numexpr (2.5)
numpy (1.10.4)
odo (0.4.0)
openpyxl (2.3.2)
pandas (0.17.1)
path.py (0.0.0)
patsy (0.4.0)
pep257 (0.7.0)
pep8 (1.7.0)
pexpect (3.3)
pickleshare (0.5)
Pillow (3.1.0)
pip (8.1.1)
ply (3.8)
psutil (3.4.2)
ptyprocess (0.5)
py (1.4.31)
pyasn1 (0.1.9)
pycairo (1.10.0)
pycosat (0.6.1)
pycparser (2.14)
pycrypto (2.6.1)
pycurl (7.19.5.3)
pyflakes (1.0.0)
Pygments (2.1.1)
pylama (3.0.2)
pylint (1.5.5)
pyOpenSSL (0.15.1)
pyparsing (2.0.3)
pytest (2.8.5)
python-dateutil (2.4.2)
python-telegram-bot (3.4)
pytz (2015.7)
PyYAML (3.11)
pyzmq (15.2.0)
qtconsole (4.1.1)
redis (2.10.3)
requests (2.9.1)
requirements-dev.txt (0.0.7)
rope (0.9.4)
ruamel-yaml (-VERSION)
scikit-image (0.11.3)
scikit-learn (0.17.1)
scipy (0.17.0)
setuptools (20.2.2)
simplegeneric (0.8.1)
singledispatch (3.4.0.3)
six (1.10.0)
snowballstemmer (1.2.1)
sockjs-tornado (1.0.1)
Sphinx (1.3.5)
sphinx-rtd-theme (0.1.9)
spyder (2.3.8)
SQLAlchemy (1.0.11)
statsmodels (0.6.1)
sympy (0.7.6.1)
tables (3.2.2)
terminado (0.5)
toolz (0.7.4)
tornado (4.3)
traceback2 (1.4.0)
traitlets (4.1.0)
unicodecsv (0.14.1)
unittest2 (1.1.0)
Werkzeug (0.11.3)
wheel (0.29.0)
wrapt (1.10.6)
xlrd (0.9.4)
XlsxWriter (0.8.4)
xlwt (1.0.0)
```

**Comment by eguiraud** [comment]
the fact that my pylama installation did not recognise the `--sort` option made me think I had an outdated version of pylama installed.

`pip install --upgrade pylama` updated it from version 3.0.2 to version 7.0.9, solving the problem.

I have no idea why conda installed such an old version of pylama and never updated it, but now everything works fine.
You might want to add to the documentation a minimum version requirement for pylama to avoid this sort of problems in the future.

Many thanks and keep up the good work!

**Comment by pchomik** [comment]
Great news. I will add such information in 3.0.5 version (next update)

**Comment by pchomik** [comment]
3.0.5 released
